include/device.h

```diff
@@ -6,14 +6,35 @@ namespace dronecore {
 
 class DeviceImpl;
 
+/**
+ * A Device represents a unit such as a drone or robot.
+ *
+ * A device can consists of multiple components such as an autopilot with a gimbal and camera.
+ * The device class allows to access plugins such as Action or Telemetry because it is based
+ * on DevicePluginContainer.
+ */
 class Device : public DevicePluginContainer
 {
 public:
+    /**
+     * Constructor of a device called internally.
+     *
+     * This constructor does not need to be called by any consumer of the API. Instead
+     * the device can be accessed using e.g. `DroneCore::device()`.
+     *
+     * @param impl the underlying device implementation
```

Comment:
Globally, I think that everything renders better if you assume the description is a sentence - e.g. 
```
@param impl The underlying device implementation.
```

Reply:
Ok, feel free to change these things for consistency. I think it would be great if you can go over all of these with fresh eyes.